File: src/main/java/epicsquid/superiorshields/enchantment/DamageBoostEnchantment.java
package epicsquid.superiorshields.enchantment;

import epicsquid.superiorshields.capability.shield.IShieldCapability;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.enchantment.Enchantment;
import net.minecraft.world.item.enchantment.EnchantmentCategory;

public abstract class DamageBoostEnchantment extends ShieldEnchantment {

	public DamageBoostEnchantment(Rarity rarityIn, EnchantmentCategory typeIn) {
		super(rarityIn, typeIn);
	}

	@Override
	protected boolean checkCompatibility(Enchantment ench) {
		return !(ench instanceof DamageBoostEnchantment);
	}

	public abstract float boostDamage(float damage);

	public abstract boolean shouldBoostDamage(IShieldCapability shield);
}

File: src/main/kotlin/dev/epicsquid/superiorshields/enchantment/DamageBoostEnchantment.kt
package dev.epicsquid.superiorshields.enchantment

import dev.epicsquid.superiorshields.effects.EffectHandler
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.item.enchantment.Enchantment
import net.minecraft.world.item.enchantment.EnchantmentCategory

abstract class DamageBoostEnchantment(
	rarity: Rarity,
	category: EnchantmentCategory,
	effectHandler: EffectHandler
) : SuperiorShieldEnchantment(rarity, category, effectHandler) {

	override fun checkCompatibility(other: Enchantment): Boolean =
		other !is DamageBoostEnchantment

	abstract fun boostDamage(damage: Float): Float

	abstract fun shouldBoostDamage(entity: LivingEntity): Boolean
}